### PR TITLE
core/vm: refactor push-functions to use `min` builtin

### PR DESCRIPTION
**Pull request by itsdevbear**

### CODE REVIEW

**Review by holiman — CHANGES_REQUESTED** [review]
The changes to `makePush` are fine, but I don't see why you made changes to `opPush1`?

**Comment by itsdevbear** [comment]
@holiman reverted, though could reduce the lines of code on push1 if we want, integer and codeLen only used once.

**Review by holiman — APPROVED** [review]
LGTM

File: core/vm/instructions.go [hunk 1]
```diff
@@ -889,22 +889,17 @@ func opPush1(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]by
...
-		}
-
-		integer := new(uint256.Int)
-		scope.Stack.push(integer.SetBytes(common.RightPadBytes(
-			scope.Contract.Code[startMin:endMin], pushByteSize)))
-
+		var (
+			codeLen = len(scope.Contract.Code)
+			start   = min(codeLen, int(*pc+1))
+			end     = min(start+pushByteSize, codeLen)
```
Comment (karalabe) [inline core/vm/instructions.go:895]: Could you flip the args to keep it consistent with the old code (and the pre min invocation?)

**Review by karalabe — APPROVED** [review]
LGTM

**Pull request merged**